keyboards/zlabkeeb/vrynboard/vrynboard.c

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2024 zlabkeeb (zlabkeeb@gmail.com)
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "zlabkeeb.h"
+
+const uint8_t max_layer = 3;
+uint8_t current_display_mode = 0;
+
+bool key_pressed = false;
+
+#ifdef OLED_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+  return OLED_ROTATION_180;
+}
+#endif
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  if (!process_record_user(keycode, record)) {
+    return false;
+  }
+#ifdef OLED_ENABLE
+  key_pressed = record->event.pressed;
+#endif
+  switch (keycode) {
+    case LT(0, KC_NO):
```

Comment:
This keycode should named something different to prevent confusion/misuse, as it is not functionally valid.

Reply:
@waffle87 

please give advice
I don't know what else to do

Reply:
Something like
```c
// zlabkeeb.h
enum custom_keycodes {
    OLED_LAYER_UPDATE = QK_KB_0
};

// zlabkeeb.c
bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
    // ...
    switch (keycode) {
        case OLED_LAYER_UPDATE:
            // ...
    }
    return false;
}
```
Or a more preferable option is to use existing [layer keycodes](https://docs.qmk.fm/feature_layers#switching-and-toggling-layers) and render the appropriate OLED data in `layer_state_set_kb` function.

Reply:
okay
I will try it

Reply:
when I use 
case OLED_LAYER_UPDATE:
I can't do on tap &  on hold

 // on tap
                if (record->tap.count) {
                    if (get_highest_layer(layer_state) >= max_layer) {
                        layer_clear();
                    } else {
                        layer_move(get_highest_layer(layer_state) + 1);
                    }
                }
#ifdef OLED_ENABLE
                // on hold
                else {
                    current_display_mode = (current_display_mode + 1) % 3;
                }

Reply:
//oled.c

bool oled_task_kb(void) {
     oled_on();
    if (!oled_task_user()) {
        return false;
    }
    if (!logo_rendered) {
        render_logo();
        if (timer_read() > 5000)
            logo_rendered = true;
    }
    else {
        switch(current_display_mode) {
            case 0:
                render_layers();
                break;
            case 1:
                render_animation();
                break;
            case 2:
                render_button_pressed();
                break;
        }
    }

        return false;
}
#endif

Reply:
update done !!